src/Camera.cpp

```diff
@@ -555,3 +555,50 @@ void Camera::PrincipalShadows(const Body *b, const int n, std::vector<Shadow> &s
 		shadowsOut.push_back(*(it++));
 	}
 }
+void Camera::CalcInteriorLighting(const Body* b, Color4ub &sLight, double &sFac) const
+{
+	for(const auto& ss : m_spaceStations)
+	{
+		SpaceStation* as_ss = (SpaceStation*)ss;
+		// Only one of these calls will do anything
+		if(as_ss->CalcInteriorLighting(b, sLight, sFac))
+			return;
```

Comment:
This function should perform early-out rejection if the distance between the body and the space station is less than `b->GetClipRadius() + ss->GetClipRadius()`.